fission-core/src/authority/error.rs

```diff
@@ -0,0 +1,54 @@
+//! Authority error type and implementations
+
+use axum::{
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    Json,
+};
+
+use serde_json::json;
+
+///////////
+// TYPES //
+///////////
+
+#[derive(Debug)]
+/// Implements https://github.com/ucan-wg/ucan-as-bearer-token#33-errors
+pub enum Error {
+    InsufficientCapabilityScope,
+    InvalidUcan { reason: String },
+    MissingCredentials,
+    MissingProofs { proofs_needed: Vec<String> },
+}
+
+/////////////////////
+// IMPLEMENTATIONS //
+/////////////////////
+
+impl IntoResponse for Error {
+    fn into_response(self) -> Response {
+        let note;
+
+        let (status, json_value) = match self {
+            Error::InsufficientCapabilityScope => (
+                StatusCode::FORBIDDEN,
+                json!({ "error": "Insufficient capability scope" }),
+            ),
+
+            Error::InvalidUcan { reason } => {
+                note = format!("Invalid UCAN: {}", reason);
+                (StatusCode::UNAUTHORIZED, json!({ "error": note.as_str() }))
```

Comment:
@zeeshanlakhani To get this to work I had to declare the note variable outside the scope of `match self`. Is there a way around this?

Reply:
Did you have to make this a reference here, via `as_str`? I'd just directly pop in the format:

```rust
json!({"error": format!("Invalid UCAN: {reason}")})
```

Reply:
Huh.. why didn't I think of that? 😄👍 I guess I didn't try that because `format!` returns a `String` not a `&str` 🤔